### PR TITLE
Add Google Analytics event tracking for item views

### PR DESCRIPTION
**Pull request by AudreyAltman**
This adds Google Analytics event tracking for views of full-sized items in exhibitions.

There are two sources of metadata for any item: the Omeka database and the DPLA API.  The Omeka database stores information that the curators enter through the dashboard.  The purpose of curator-provided metadata is threefold:
1. Curators can use fields that aren't available through the DPLA API, such as formatted citations.
2. Curators can override fields that may have opaque or poorly formatted data in the API, such as titles.
3. Curators can provide data for items that are not in the DPLA API.  Sometimes they get special permission to use an item we don't have through the API, and sometimes items that were initially available through the API are no longer there.

The existing `ItemMetadata` class provides functions for getting item metadata from either the Omeka database or the API.  It's existing functionality privileges data from Omeka (which supports #2 above) and falls back to data from the API if Omeka data is unavailable.  For the purposes of collecting and aggregating usage stats, we want the opposite - API data if it as available, then fall back to Omeka data. This PR adds an optional parameter to relevant functions to indicate a preference for API data.

Item metadata is made available to JavaScript functions through HTML `data-*` attributes.  This follows the convention used in other frontend applications.  The beacons sent to Google Analytics via JavaScript also follow a format used in other apps:

```
Category: "View Exhibition Item : [provider name]"
Action: "[data provider name]"
Label: "[item id] : [item title]"
```

Some pages have slideshows with multiple items, in which case all items in the slideshow are tracked as views when the page loads.  According to FA, it would be nice to track items one by one as they are viewed, but not required.  Due to time restraints I am not pursuing that feature in this PR.

This has been testing on staging using the Google Analytics Debugger plugin.  It addresses [ticket #8441](https://issues.dp.la/issues/8441).

### CODE REVIEW

File: plugins/ExhibitBuilder/helpers/DPLAFunctions.php [hunk 1]
```diff
@@ -644,19 +669,24 @@ function get_id() {
...
-    private function get_field_value($omeka_field_name, $api_field_name) {
+    private function get_field_value($omeka_field_name, $api_field_name, $api_preferred = false) {
 
         $omeka_field_value = dpla_get_field_value_by_name($this->item, $omeka_field_name);
         $api_field_value = $this->json ? 
             dpla_get_field_value_by_arrayname($this->json, $api_field_name) : null;
+        $field_value = null;
 
-        $field_value = $omeka_field_value ?: $api_field_value;
+        if ($api_preferred == false) {
```
Comment (mdellabitta) [inline plugins/ExhibitBuilder/helpers/DPLAFunctions.php]: The way this is written, it seems like more than one call gets made to get the field value either way. It seems like it'd be a little better to make whatever the first call is, and only fall back to the second if the first is unavailable.

**Comment by AudreyAltman** [comment]
@mdellabitta I _finally_ have a working refactor that lazy loads API calls, only calling the API if necessary.  Before, the API was called when the class was instantiated.

I wasn't able to find an elegant design pattern for lazy loading a private class variable -- I got hung up for a while trying to implement the magic `__get` method before I realized that it doesn't execute for a private class variable from within the same class (instead, the variable is just directly accessed).  So instead, throughout the class, I substituted a direct call to `$this->json` with a call to `$this->get_json()` The `get_json()` method lazy loads `$json` with an API call.

I also tweaked the logic flow in `private function get_field_value` to ensure that `get_json()` is only called if necessary.

I can squash after you get a chance to review.

**Comment by mdellabitta** [comment]
Yeah, hiding the lazy loading behind a method call is definitely the right way to go.

I wonder if you couldn't have hidden the logic for the field lookup behind a private helper function as well, so you don't have to add branches on $this->get_json() != false in various places.

**Comment by AudreyAltman** [comment]
Good idea @mdellabitta !  I'll do that.

**Comment by AudreyAltman** [comment]
@mdellabitta okay, helper function has been added.  Also I moved all the private methods to the bottom of the class to tidy up.  I tested on staging to confirm that everything still works.

**Pull request merged**